scipy/stats/_stats_py.py

```diff
@@ -617,14 +621,18 @@ def tmean(a, limits=None, inclusive=(True, True), axis=None):
     10.0
 
     """
-    a = asarray(a)
+    if not np.issubdtype(a.dtype, np.inexact):
+        a = a.astype(np.float64)
     if limits is None:
         return np.mean(a, axis)
     am = _mask_to_limits(a, limits, inclusive)
-    mean = np.ma.filled(am.mean(axis=axis), fill_value=np.nan)
-    return mean if mean.ndim > 0 else mean.item()
+    amnan = am.filled(fill_value=_get_nan(a))
```

Comment:
It's probably ok to use just `np.nan` since it will convert to the right dtype when you put it in the array.

Reply:
Sure, will change that.

Reply:
Done.